CPAC/pipeline/engine.py

```diff
@@ -108,6 +108,34 @@ def __init__(self, rpool=None, name=None, cfg=None, pipe_list=None):
     def append_name(self, name):
         self.name.append(name)
 
+    def back_propogate_template_name(self, resource_idx: str, json_info: dict,
+                                     id_string: 'pe.Node') -> None:
+        """Find and apply the template name from a resource's provenance
+
+        Parameters
+        ----------
+        resource_idx : str
+
+        json_info : dict
+
+        id_string : pe.Node
+
+        Returns
+        -------
+        None
+        """
+        if 'Template' in json_info:
+            id_string.inputs.template_desc = json_info['Template']
+        elif ('space-template' in resource_idx and
+              len(json_info.get('CpacProvenance', [])) > 1):
+            try:
+                # check parent resource
+                self.back_propogate_template_name(resource_idx,
+                    list(self.get(json_info['CpacProvenance'][-2][-1].split(
+                        ':')[0]).values())[0]['json'], id_string)
```

Comment:
@sgiavasis is `json_info['CpacProvenance'][-2][-1].split(':')[0]` too specific?

It works for the ones I tested it on, but I'm not confident that's always the (only) item we care to check as we climb the provenance tree

Reply:
What does this evaluate to exactly?

Reply:
Here's an example:

#### `resource_idx`
```Python
'space-template_desc-sm_alff'
```

#### `list(self.get(json_info['CpacProvenance'][-2][-1].split(':')[0]).values())[0]['json']`
```Python
{'CpacProvenance': [[[[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], [[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['TR:func_metadata_ingress'], ['tpattern:func_metadata_ingress'], 'desc-preproc_bold:func_slice_time'], [['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], 'desc-preproc_bold:motion_correction_only_mcflirt'], ['FSL-AFNI-bold-ref:template_resample'], ['FSL-AFNI-brain-mask:template_resample'], ['FSL-AFNI-brain-probseg:template_resample'], 'desc-ref_bold:bold_mask_fsl_afni', 'sbref:coregistration_prep_fmriprep'], [[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], [[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['TR:func_metadata_ingress'], ['tpattern:func_metadata_ingress'], 'desc-preproc_bold:func_slice_time'], [['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], 'desc-preproc_bold:motion_correction_only_mcflirt'], ['FSL-AFNI-bold-ref:template_resample'], ['FSL-AFNI-brain-mask:template_resample'], ['FSL-AFNI-brain-probseg:template_resample'], 'space-bold_desc-brain_mask:bold_mask_fsl_afni'], ['pe-direction:func_metadata_ingress'], ['epi-1:fmap_ingress'], ['epi-1-pedir:fmap_pedir_ingress'], ['epi-1-dwell:fmap_dwell_ingress'], ['epi-1-total-readout:fmap_readout_ingress'], ['epi-2:fmap_ingress'], ['epi-2-pedir:fmap_pedir_ingress'], ['epi-2-dwell:fmap_dwell_ingress'], ['epi-2-total-readout:fmap_readout_ingress'], 'sbref:distcor_blip_fsl_topup'], [[[[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], [[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['TR:func_metadata_ingress'], ['tpattern:func_metadata_ingress'], 'desc-preproc_bold:func_slice_time'], [['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], 'desc-preproc_bold:motion_correction_only_mcflirt'], ['FSL-AFNI-bold-ref:template_resample'], ['FSL-AFNI-brain-mask:template_resample'], ['FSL-AFNI-brain-probseg:template_resample'], 'desc-ref_bold:bold_mask_fsl_afni', 'sbref:coregistration_prep_fmriprep'], [[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], [[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['TR:func_metadata_ingress'], ['tpattern:func_metadata_ingress'], 'desc-preproc_bold:func_slice_time'], [['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], 'desc-preproc_bold:motion_correction_only_mcflirt'], ['FSL-AFNI-bold-ref:template_resample'], ['FSL-AFNI-brain-mask:template_resample'], ['FSL-AFNI-brain-probseg:template_resample'], 'space-bold_desc-brain_mask:bold_mask_fsl_afni'], ['pe-direction:func_metadata_ingress'], ['epi-1:fmap_ingress'], ['epi-1-pedir:fmap_pedir_ingress'], ['epi-1-dwell:fmap_dwell_ingress'], ['epi-1-total-readout:fmap_readout_ingress'], ['epi-2:fmap_ingress'], ['epi-2-pedir:fmap_pedir_ingress'], ['epi-2-dwell:fmap_dwell_ingress'], ['epi-2-total-readout:fmap_readout_ingress'], 'sbref:distcor_blip_fsl_topup'], [[[[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], [[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['TR:func_metadata_ingress'], ['tpattern:func_metadata_ingress'], 'desc-preproc_bold:func_slice_time'], [['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], 'desc-preproc_bold:motion_correction_only_mcflirt'], ['FSL-AFNI-bold-ref:template_resample'], ['FSL-AFNI-brain-mask:template_resample'], ['FSL-AFNI-brain-probseg:template_resample'], 'desc-ref_bold:bold_mask_fsl_afni', 'sbref:coregistration_prep_fmriprep'], [[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], [[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['TR:func_metadata_ingress'], ['tpattern:func_metadata_ingress'], 'desc-preproc_bold:func_slice_time'], [['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], 'desc-preproc_bold:motion_correction_only_mcflirt'], ['FSL-AFNI-bold-ref:template_resample'], ['FSL-AFNI-brain-mask:template_resample'], ['FSL-AFNI-brain-probseg:template_resample'], 'space-bold_desc-brain_mask:bold_mask_fsl_afni'], ['pe-direction:func_metadata_ingress'], ['epi-1:fmap_ingress'], ['epi-1-pedir:fmap_pedir_ingress'], ['epi-1-dwell:fmap_dwell_ingress'], ['epi-1-total-readout:fmap_readout_ingress'], ['epi-2:fmap_ingress'], ['epi-2-pedir:fmap_pedir_ingress'], ['epi-2-dwell:fmap_dwell_ingress'], ['epi-2-total-readout:fmap_readout_ingress'], 'sbref:distcor_blip_fsl_topup'], [[[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['TR:func_metadata_ingress'], ['tpattern:func_metadata_ingress'], 'desc-preproc_bold:func_slice_time'], [['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], 'desc-preproc_bold:motion_correction_only_mcflirt'], [[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], [[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['TR:func_metadata_ingress'], ['tpattern:func_metadata_ingress'], 'desc-preproc_bold:func_slice_time'], [['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], 'desc-preproc_bold:motion_correction_only_mcflirt'], ['FSL-AFNI-bold-ref:template_resample'], ['FSL-AFNI-brain-mask:template_resample'], ['FSL-AFNI-brain-probseg:template_resample'], 'space-bold_desc-brain_mask:bold_mask_fsl_afni'], 'desc-preproc_bold:bold_masking'], [['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['TR:func_metadata_ingress'], ['tpattern:func_metadata_ingress'], 'desc-stc_bold:func_slice_time'], ['bold:func_ingress'], [[[[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], [[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['TR:func_metadata_ingress'], ['tpattern:func_metadata_ingress'], 'desc-preproc_bold:func_slice_time'], [['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], 'desc-preproc_bold:motion_correction_only_mcflirt'], ['FSL-AFNI-bold-ref:template_resample'], ['FSL-AFNI-brain-mask:template_resample'], ['FSL-AFNI-brain-probseg:template_resample'], 'desc-ref_bold:bold_mask_fsl_afni', 'sbref:coregistration_prep_fmriprep'], [[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], [[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['TR:func_metadata_ingress'], ['tpattern:func_metadata_ingress'], 'desc-preproc_bold:func_slice_time'], [['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], 'desc-preproc_bold:motion_correction_only_mcflirt'], ['FSL-AFNI-bold-ref:template_resample'], ['FSL-AFNI-brain-mask:template_resample'], ['FSL-AFNI-brain-probseg:template_resample'], 'space-bold_desc-brain_mask:bold_mask_fsl_afni'], ['pe-direction:func_metadata_ingress'], ['epi-1:fmap_ingress'], ['epi-1-pedir:fmap_pedir_ingress'], ['epi-1-dwell:fmap_dwell_ingress'], ['epi-1-total-readout:fmap_readout_ingress'], ['epi-2:fmap_ingress'], ['epi-2-pedir:fmap_pedir_ingress'], ['epi-2-dwell:fmap_dwell_ingress'], ['epi-2-total-readout:fmap_readout_ingress'], 'sbref:distcor_blip_fsl_topup'], [[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['TR:func_metadata_ingress'], ['tpattern:func_metadata_ingress'], 'desc-preproc_bold:func_slice_time'], [['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], 'desc-motion_bold:motion_correction_only_mcflirt'], ['pe-direction:func_metadata_ingress'], [['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'desc-preproc_T1w:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-head_T1w:anatomical_init'], 'desc-preproc_T1w:brain_extraction'], ['T2w:anat_ingress'], [[['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'desc-preproc_T1w:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-head_T1w:anatomical_init'], 'desc-brain_T1w:brain_extraction'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], [[['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'desc-preproc_T1w:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-head_T1w:anatomical_init'], 'desc-preproc_T1w:brain_extraction'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], [['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'desc-preproc_T1w:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-head_T1w:anatomical_init'], 'desc-head_T1w:brain_extraction'], ['T1w-template:template_resample'], ['T1w-brain-template:template_resample'], ['T1w-brain-template-mask:template_resample'], 'from-template_to-T1w_mode-image_desc-linear_xfm:register_ANTs_anat_to_template'], ['CSF-path:CSF-path_config_ingress'], ['GM-path:GM-path_config_ingress'], ['WM-path:WM-path_config_ingress'], 'label-WM_probseg:tissue_seg_fsl_fast'], [[['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'desc-preproc_T1w:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-head_T1w:anatomical_init'], 'desc-brain_T1w:brain_extraction'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], [[['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'desc-preproc_T1w:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-head_T1w:anatomical_init'], 'desc-preproc_T1w:brain_extraction'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], [['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'desc-preproc_T1w:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-head_T1w:anatomical_init'], 'desc-head_T1w:brain_extraction'], ['T1w-template:template_resample'], ['T1w-brain-template:template_resample'], ['T1w-brain-template-mask:template_resample'], 'from-template_to-T1w_mode-image_desc-linear_xfm:register_ANTs_anat_to_template'], ['CSF-path:CSF-path_config_ingress'], ['GM-path:GM-path_config_ingress'], ['WM-path:WM-path_config_ingress'], 'label-WM_pveseg:tissue_seg_fsl_fast'], [['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'desc-preproc_T1w:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-head_T1w:anatomical_init'], 'desc-head_T1w:brain_extraction'], 'from-bold_to-T1w_mode-image_desc-linear_xfm:coregistration'], ['pe-direction:func_metadata_ingress'], 'desc-preproc_bold:apply_phasediff_to_timeseries_separately_single_step_resampling_from_stc'], [['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['TR:func_metadata_ingress'], ['tpattern:func_metadata_ingress'], 'desc-stc_bold:func_slice_time'], ['bold:func_ingress'], [[[[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], [[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['TR:func_metadata_ingress'], ['tpattern:func_metadata_ingress'], 'desc-preproc_bold:func_slice_time'], [['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], 'desc-preproc_bold:motion_correction_only_mcflirt'], ['FSL-AFNI-bold-ref:template_resample'], ['FSL-AFNI-brain-mask:template_resample'], ['FSL-AFNI-brain-probseg:template_resample'], 'desc-ref_bold:bold_mask_fsl_afni', 'sbref:coregistration_prep_fmriprep'], [[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], [[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['TR:func_metadata_ingress'], ['tpattern:func_metadata_ingress'], 'desc-preproc_bold:func_slice_time'], [['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], 'desc-preproc_bold:motion_correction_only_mcflirt'], ['FSL-AFNI-bold-ref:template_resample'], ['FSL-AFNI-brain-mask:template_resample'], ['FSL-AFNI-brain-probseg:template_resample'], 'space-bold_desc-brain_mask:bold_mask_fsl_afni'], ['pe-direction:func_metadata_ingress'], ['epi-1:fmap_ingress'], ['epi-1-pedir:fmap_pedir_ingress'], ['epi-1-dwell:fmap_dwell_ingress'], ['epi-1-total-readout:fmap_readout_ingress'], ['epi-2:fmap_ingress'], ['epi-2-pedir:fmap_pedir_ingress'], ['epi-2-dwell:fmap_dwell_ingress'], ['epi-2-total-readout:fmap_readout_ingress'], 'sbref:distcor_blip_fsl_topup'], [[[[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], [[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['TR:func_metadata_ingress'], ['tpattern:func_metadata_ingress'], 'desc-preproc_bold:func_slice_time'], [['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], 'desc-preproc_bold:motion_correction_only_mcflirt'], ['FSL-AFNI-bold-ref:template_resample'], ['FSL-AFNI-brain-mask:template_resample'], ['FSL-AFNI-brain-probseg:template_resample'], 'desc-ref_bold:bold_mask_fsl_afni', 'sbref:coregistration_prep_fmriprep'], [[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], [[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['TR:func_metadata_ingress'], ['tpattern:func_metadata_ingress'], 'desc-preproc_bold:func_slice_time'], [['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], 'desc-preproc_bold:motion_correction_only_mcflirt'], ['FSL-AFNI-bold-ref:template_resample'], ['FSL-AFNI-brain-mask:template_resample'], ['FSL-AFNI-brain-probseg:template_resample'], 'space-bold_desc-brain_mask:bold_mask_fsl_afni'], ['pe-direction:func_metadata_ingress'], ['epi-1:fmap_ingress'], ['epi-1-pedir:fmap_pedir_ingress'], ['epi-1-dwell:fmap_dwell_ingress'], ['epi-1-total-readout:fmap_readout_ingress'], ['epi-2:fmap_ingress'], ['epi-2-pedir:fmap_pedir_ingress'], ['epi-2-dwell:fmap_dwell_ingress'], ['epi-2-total-readout:fmap_readout_ingress'], 'sbref:distcor_blip_fsl_topup'], [[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['TR:func_metadata_ingress'], ['tpattern:func_metadata_ingress'], 'desc-preproc_bold:func_slice_time'], [['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], 'desc-motion_bold:motion_correction_only_mcflirt'], ['pe-direction:func_metadata_ingress'], [['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'desc-preproc_T1w:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-head_T1w:anatomical_init'], 'desc-preproc_T1w:brain_extraction'], ['T2w:anat_ingress'], [[['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'desc-preproc_T1w:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-head_T1w:anatomical_init'], 'desc-brain_T1w:brain_extraction'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], [[['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'desc-preproc_T1w:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-head_T1w:anatomical_init'], 'desc-preproc_T1w:brain_extraction'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], [['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'desc-preproc_T1w:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-head_T1w:anatomical_init'], 'desc-head_T1w:brain_extraction'], ['T1w-template:template_resample'], ['T1w-brain-template:template_resample'], ['T1w-brain-template-mask:template_resample'], 'from-template_to-T1w_mode-image_desc-linear_xfm:register_ANTs_anat_to_template'], ['CSF-path:CSF-path_config_ingress'], ['GM-path:GM-path_config_ingress'], ['WM-path:WM-path_config_ingress'], 'label-WM_probseg:tissue_seg_fsl_fast'], [[['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'desc-preproc_T1w:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-head_T1w:anatomical_init'], 'desc-brain_T1w:brain_extraction'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], [[['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'desc-preproc_T1w:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-head_T1w:anatomical_init'], 'desc-preproc_T1w:brain_extraction'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], [['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'desc-preproc_T1w:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-head_T1w:anatomical_init'], 'desc-head_T1w:brain_extraction'], ['T1w-template:template_resample'], ['T1w-brain-template:template_resample'], ['T1w-brain-template-mask:template_resample'], 'from-template_to-T1w_mode-image_desc-linear_xfm:register_ANTs_anat_to_template'], ['CSF-path:CSF-path_config_ingress'], ['GM-path:GM-path_config_ingress'], ['WM-path:WM-path_config_ingress'], 'label-WM_pveseg:tissue_seg_fsl_fast'], [['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'desc-preproc_T1w:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-head_T1w:anatomical_init'], 'desc-head_T1w:brain_extraction'], 'from-bold_to-T1w_mode-image_desc-linear_xfm:coregistration'], [[[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], [[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['TR:func_metadata_ingress'], ['tpattern:func_metadata_ingress'], 'desc-preproc_bold:func_slice_time'], [['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], 'desc-preproc_bold:motion_correction_only_mcflirt'], ['FSL-AFNI-bold-ref:template_resample'], ['FSL-AFNI-brain-mask:template_resample'], ['FSL-AFNI-brain-probseg:template_resample'], 'desc-ref_bold:bold_mask_fsl_afni', 'sbref:coregistration_prep_fmriprep'], [[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], [[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['TR:func_metadata_ingress'], ['tpattern:func_metadata_ingress'], 'desc-preproc_bold:func_slice_time'], [['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], 'desc-preproc_bold:motion_correction_only_mcflirt'], ['FSL-AFNI-bold-ref:template_resample'], ['FSL-AFNI-brain-mask:template_resample'], ['FSL-AFNI-brain-probseg:template_resample'], 'space-bold_desc-brain_mask:bold_mask_fsl_afni'], ['pe-direction:func_metadata_ingress'], ['epi-1:fmap_ingress'], ['epi-1-pedir:fmap_pedir_ingress'], ['epi-1-dwell:fmap_dwell_ingress'], ['epi-1-total-readout:fmap_readout_ingress'], ['epi-2:fmap_ingress'], ['epi-2-pedir:fmap_pedir_ingress'], ['epi-2-dwell:fmap_dwell_ingress'], ['epi-2-total-readout:fmap_readout_ingress'], 'fsl-blip-warp:distcor_blip_fsl_topup'], [[['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'desc-preproc_T1w:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-head_T1w:anatomical_init'], 'desc-preproc_T1w:brain_extraction'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], [['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'desc-preproc_T1w:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-head_T1w:anatomical_init'], 'desc-head_T1w:brain_extraction'], ['T1w-template:template_resample'], ['T1w-brain-template:template_resample'], ['T1w-brain-template-mask:template_resample'], 'from-T1w_to-template_mode-image_xfm:register_ANTs_anat_to_template'], [[['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'desc-preproc_T1w:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-head_T1w:anatomical_init'], 'desc-preproc_T1w:brain_extraction'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], [['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'desc-preproc_T1w:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-head_T1w:anatomical_init'], 'desc-head_T1w:brain_extraction'], ['T1w-template:template_resample'], ['T1w-brain-template:template_resample'], ['T1w-brain-template-mask:template_resample'], 'from-template_to-T1w_mode-image_xfm:register_ANTs_anat_to_template'], [['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'desc-preproc_T1w:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-head_T1w:anatomical_init'], 'desc-brain_T1w:brain_extraction'], ['T1w-brain-template-funcreg:template_resample'], 'from-bold_to-template_mode-image_xfm:create_func_to_T1template_xfm'], [[[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], [[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['TR:func_metadata_ingress'], ['tpattern:func_metadata_ingress'], 'desc-preproc_bold:func_slice_time'], [['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], 'desc-preproc_bold:motion_correction_only_mcflirt'], ['FSL-AFNI-bold-ref:template_resample'], ['FSL-AFNI-brain-mask:template_resample'], ['FSL-AFNI-brain-probseg:template_resample'], 'desc-ref_bold:bold_mask_fsl_afni', 'sbref:coregistration_prep_fmriprep'], [[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], [[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['TR:func_metadata_ingress'], ['tpattern:func_metadata_ingress'], 'desc-preproc_bold:func_slice_time'], [['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], 'desc-preproc_bold:motion_correction_only_mcflirt'], ['FSL-AFNI-bold-ref:template_resample'], ['FSL-AFNI-brain-mask:template_resample'], ['FSL-AFNI-brain-probseg:template_resample'], 'space-bold_desc-brain_mask:bold_mask_fsl_afni'], ['pe-direction:func_metadata_ingress'], ['epi-1:fmap_ingress'], ['epi-1-pedir:fmap_pedir_ingress'], ['epi-1-dwell:fmap_dwell_ingress'], ['epi-1-total-readout:fmap_readout_ingress'], ['epi-2:fmap_ingress'], ['epi-2-pedir:fmap_pedir_ingress'], ['epi-2-dwell:fmap_dwell_ingress'], ['epi-2-total-readout:fmap_readout_ingress'], 'fsl-blip-warp:distcor_blip_fsl_topup'], 'desc-stc_bold:apply_blip_to_timeseries_separately_single_step_resampling_from_stc'], [['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], [[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], [[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['TR:func_metadata_ingress'], ['tpattern:func_metadata_ingress'], 'desc-preproc_bold:func_slice_time'], [['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], 'desc-preproc_bold:motion_correction_only_mcflirt'], ['FSL-AFNI-bold-ref:template_resample'], ['FSL-AFNI-brain-mask:template_resample'], ['FSL-AFNI-brain-probseg:template_resample'], 'space-bold_desc-brain_mask:bold_mask_fsl_afni'], [['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], [['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], 'coordinate-transformation:motion_estimates_mcflirt'], [[['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'desc-preproc_T1w:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-head_T1w:anatomical_init'], 'desc-preproc_T1w:brain_extraction'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], [['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'desc-preproc_T1w:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-head_T1w:anatomical_init'], 'desc-head_T1w:brain_extraction'], ['T1w-template:template_resample'], ['T1w-brain-template:template_resample'], ['T1w-brain-template-mask:template_resample'], 'from-T1w_to-template_mode-image_xfm:register_ANTs_anat_to_template'], [[[[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], [[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['TR:func_metadata_ingress'], ['tpattern:func_metadata_ingress'], 'desc-preproc_bold:func_slice_time'], [['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], 'desc-preproc_bold:motion_correction_only_mcflirt'], ['FSL-AFNI-bold-ref:template_resample'], ['FSL-AFNI-brain-mask:template_resample'], ['FSL-AFNI-brain-probseg:template_resample'], 'desc-ref_bold:bold_mask_fsl_afni', 'sbref:coregistration_prep_fmriprep'], [[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], [[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['TR:func_metadata_ingress'], ['tpattern:func_metadata_ingress'], 'desc-preproc_bold:func_slice_time'], [['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], 'desc-preproc_bold:motion_correction_only_mcflirt'], ['FSL-AFNI-bold-ref:template_resample'], ['FSL-AFNI-brain-mask:template_resample'], ['FSL-AFNI-brain-probseg:template_resample'], 'space-bold_desc-brain_mask:bold_mask_fsl_afni'], ['pe-direction:func_metadata_ingress'], ['epi-1:fmap_ingress'], ['epi-1-pedir:fmap_pedir_ingress'], ['epi-1-dwell:fmap_dwell_ingress'], ['epi-1-total-readout:fmap_readout_ingress'], ['epi-2:fmap_ingress'], ['epi-2-pedir:fmap_pedir_ingress'], ['epi-2-dwell:fmap_dwell_ingress'], ['epi-2-total-readout:fmap_readout_ingress'], 'sbref:distcor_blip_fsl_topup'], [[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['TR:func_metadata_ingress'], ['tpattern:func_metadata_ingress'], 'desc-preproc_bold:func_slice_time'], [['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], 'desc-motion_bold:motion_correction_only_mcflirt'], ['pe-direction:func_metadata_ingress'], [['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'desc-preproc_T1w:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-head_T1w:anatomical_init'], 'desc-preproc_T1w:brain_extraction'], ['T2w:anat_ingress'], [[['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'desc-preproc_T1w:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-head_T1w:anatomical_init'], 'desc-brain_T1w:brain_extraction'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], [[['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'desc-preproc_T1w:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-head_T1w:anatomical_init'], 'desc-preproc_T1w:brain_extraction'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], [['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'desc-preproc_T1w:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-head_T1w:anatomical_init'], 'desc-head_T1w:brain_extraction'], ['T1w-template:template_resample'], ['T1w-brain-template:template_resample'], ['T1w-brain-template-mask:template_resample'], 'from-template_to-T1w_mode-image_desc-linear_xfm:register_ANTs_anat_to_template'], ['CSF-path:CSF-path_config_ingress'], ['GM-path:GM-path_config_ingress'], ['WM-path:WM-path_config_ingress'], 'label-WM_probseg:tissue_seg_fsl_fast'], [[['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'desc-preproc_T1w:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-head_T1w:anatomical_init'], 'desc-brain_T1w:brain_extraction'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], [[['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'desc-preproc_T1w:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-head_T1w:anatomical_init'], 'desc-preproc_T1w:brain_extraction'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], [['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'desc-preproc_T1w:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-head_T1w:anatomical_init'], 'desc-head_T1w:brain_extraction'], ['T1w-template:template_resample'], ['T1w-brain-template:template_resample'], ['T1w-brain-template-mask:template_resample'], 'from-template_to-T1w_mode-image_desc-linear_xfm:register_ANTs_anat_to_template'], ['CSF-path:CSF-path_config_ingress'], ['GM-path:GM-path_config_ingress'], ['WM-path:WM-path_config_ingress'], 'label-WM_pveseg:tissue_seg_fsl_fast'], [['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'desc-preproc_T1w:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-head_T1w:anatomical_init'], 'desc-head_T1w:brain_extraction'], 'from-bold_to-T1w_mode-image_desc-linear_xfm:coregistration'], [[[[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], [[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['TR:func_metadata_ingress'], ['tpattern:func_metadata_ingress'], 'desc-preproc_bold:func_slice_time'], [['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], 'desc-preproc_bold:motion_correction_only_mcflirt'], ['FSL-AFNI-bold-ref:template_resample'], ['FSL-AFNI-brain-mask:template_resample'], ['FSL-AFNI-brain-probseg:template_resample'], 'desc-ref_bold:bold_mask_fsl_afni', 'sbref:coregistration_prep_fmriprep'], [[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], [[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['TR:func_metadata_ingress'], ['tpattern:func_metadata_ingress'], 'desc-preproc_bold:func_slice_time'], [['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], 'desc-preproc_bold:motion_correction_only_mcflirt'], ['FSL-AFNI-bold-ref:template_resample'], ['FSL-AFNI-brain-mask:template_resample'], ['FSL-AFNI-brain-probseg:template_resample'], 'space-bold_desc-brain_mask:bold_mask_fsl_afni'], ['pe-direction:func_metadata_ingress'], ['epi-1:fmap_ingress'], ['epi-1-pedir:fmap_pedir_ingress'], ['epi-1-dwell:fmap_dwell_ingress'], ['epi-1-total-readout:fmap_readout_ingress'], ['epi-2:fmap_ingress'], ['epi-2-pedir:fmap_pedir_ingress'], ['epi-2-dwell:fmap_dwell_ingress'], ['epi-2-total-readout:fmap_readout_ingress'], 'sbref:distcor_blip_fsl_topup'], [[[[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], [[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['TR:func_metadata_ingress'], ['tpattern:func_metadata_ingress'], 'desc-preproc_bold:func_slice_time'], [['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], 'desc-preproc_bold:motion_correction_only_mcflirt'], ['FSL-AFNI-bold-ref:template_resample'], ['FSL-AFNI-brain-mask:template_resample'], ['FSL-AFNI-brain-probseg:template_resample'], 'desc-ref_bold:bold_mask_fsl_afni', 'sbref:coregistration_prep_fmriprep'], [[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], [[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['TR:func_metadata_ingress'], ['tpattern:func_metadata_ingress'], 'desc-preproc_bold:func_slice_time'], [['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], 'desc-preproc_bold:motion_correction_only_mcflirt'], ['FSL-AFNI-bold-ref:template_resample'], ['FSL-AFNI-brain-mask:template_resample'], ['FSL-AFNI-brain-probseg:template_resample'], 'space-bold_desc-brain_mask:bold_mask_fsl_afni'], ['pe-direction:func_metadata_ingress'], ['epi-1:fmap_ingress'], ['epi-1-pedir:fmap_pedir_ingress'], ['epi-1-dwell:fmap_dwell_ingress'], ['epi-1-total-readout:fmap_readout_ingress'], ['epi-2:fmap_ingress'], ['epi-2-pedir:fmap_pedir_ingress'], ['epi-2-dwell:fmap_dwell_ingress'], ['epi-2-total-readout:fmap_readout_ingress'], 'sbref:distcor_blip_fsl_topup'], [[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['TR:func_metadata_ingress'], ['tpattern:func_metadata_ingress'], 'desc-preproc_bold:func_slice_time'], [['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], 'desc-motion_bold:motion_correction_only_mcflirt'], ['pe-direction:func_metadata_ingress'], [['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'desc-preproc_T1w:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-head_T1w:anatomical_init'], 'desc-preproc_T1w:brain_extraction'], ['T2w:anat_ingress'], [[['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'desc-preproc_T1w:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-head_T1w:anatomical_init'], 'desc-brain_T1w:brain_extraction'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], [[['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'desc-preproc_T1w:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-head_T1w:anatomical_init'], 'desc-preproc_T1w:brain_extraction'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], [['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'desc-preproc_T1w:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-head_T1w:anatomical_init'], 'desc-head_T1w:brain_extraction'], ['T1w-template:template_resample'], ['T1w-brain-template:template_resample'], ['T1w-brain-template-mask:template_resample'], 'from-template_to-T1w_mode-image_desc-linear_xfm:register_ANTs_anat_to_template'], ['CSF-path:CSF-path_config_ingress'], ['GM-path:GM-path_config_ingress'], ['WM-path:WM-path_config_ingress'], 'label-WM_probseg:tissue_seg_fsl_fast'], [[['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'desc-preproc_T1w:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-head_T1w:anatomical_init'], 'desc-brain_T1w:brain_extraction'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], [[['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'desc-preproc_T1w:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-head_T1w:anatomical_init'], 'desc-preproc_T1w:brain_extraction'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], [['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'desc-preproc_T1w:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-head_T1w:anatomical_init'], 'desc-head_T1w:brain_extraction'], ['T1w-template:template_resample'], ['T1w-brain-template:template_resample'], ['T1w-brain-template-mask:template_resample'], 'from-template_to-T1w_mode-image_desc-linear_xfm:register_ANTs_anat_to_template'], ['CSF-path:CSF-path_config_ingress'], ['GM-path:GM-path_config_ingress'], ['WM-path:WM-path_config_ingress'], 'label-WM_pveseg:tissue_seg_fsl_fast'], [['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'desc-preproc_T1w:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-head_T1w:anatomical_init'], 'desc-head_T1w:brain_extraction'], 'from-bold_to-T1w_mode-image_desc-linear_xfm:coregistration'], [[[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], [[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['TR:func_metadata_ingress'], ['tpattern:func_metadata_ingress'], 'desc-preproc_bold:func_slice_time'], [['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], 'desc-preproc_bold:motion_correction_only_mcflirt'], ['FSL-AFNI-bold-ref:template_resample'], ['FSL-AFNI-brain-mask:template_resample'], ['FSL-AFNI-brain-probseg:template_resample'], 'desc-ref_bold:bold_mask_fsl_afni', 'sbref:coregistration_prep_fmriprep'], [[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], [[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['TR:func_metadata_ingress'], ['tpattern:func_metadata_ingress'], 'desc-preproc_bold:func_slice_time'], [['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], 'desc-preproc_bold:motion_correction_only_mcflirt'], ['FSL-AFNI-bold-ref:template_resample'], ['FSL-AFNI-brain-mask:template_resample'], ['FSL-AFNI-brain-probseg:template_resample'], 'space-bold_desc-brain_mask:bold_mask_fsl_afni'], ['pe-direction:func_metadata_ingress'], ['epi-1:fmap_ingress'], ['epi-1-pedir:fmap_pedir_ingress'], ['epi-1-dwell:fmap_dwell_ingress'], ['epi-1-total-readout:fmap_readout_ingress'], ['epi-2:fmap_ingress'], ['epi-2-pedir:fmap_pedir_ingress'], ['epi-2-dwell:fmap_dwell_ingress'], ['epi-2-total-readout:fmap_readout_ingress'], 'fsl-blip-warp:distcor_blip_fsl_topup'], [[['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'desc-preproc_T1w:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-head_T1w:anatomical_init'], 'desc-preproc_T1w:brain_extraction'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], [['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'desc-preproc_T1w:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-head_T1w:anatomical_init'], 'desc-head_T1w:brain_extraction'], ['T1w-template:template_resample'], ['T1w-brain-template:template_resample'], ['T1w-brain-template-mask:template_resample'], 'from-T1w_to-template_mode-image_xfm:register_ANTs_anat_to_template'], [[['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'desc-preproc_T1w:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-head_T1w:anatomical_init'], 'desc-preproc_T1w:brain_extraction'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], [['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'desc-preproc_T1w:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-head_T1w:anatomical_init'], 'desc-head_T1w:brain_extraction'], ['T1w-template:template_resample'], ['T1w-brain-template:template_resample'], ['T1w-brain-template-mask:template_resample'], 'from-template_to-T1w_mode-image_xfm:register_ANTs_anat_to_template'], [['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'desc-preproc_T1w:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-head_T1w:anatomical_init'], 'desc-brain_T1w:brain_extraction'], ['T1w-brain-template-funcreg:template_resample'], 'from-bold_to-template_mode-image_xfm:create_func_to_T1template_xfm'], [[[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], [[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['TR:func_metadata_ingress'], ['tpattern:func_metadata_ingress'], 'desc-preproc_bold:func_slice_time'], [['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], 'desc-preproc_bold:motion_correction_only_mcflirt'], ['FSL-AFNI-bold-ref:template_resample'], ['FSL-AFNI-brain-mask:template_resample'], ['FSL-AFNI-brain-probseg:template_resample'], 'desc-ref_bold:bold_mask_fsl_afni', 'sbref:coregistration_prep_fmriprep'], [[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], [[['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['TR:func_metadata_ingress'], ['tpattern:func_metadata_ingress'], 'desc-preproc_bold:func_slice_time'], [['bold:func_ingress', 'desc-preproc_bold:func_reorient', 'desc-preproc_bold:func_truncate'], ['bold:func_ingress', 'desc-reorient_bold:func_reorient'], 'motion-basefile:get_motion_ref_fmriprep_reference'], 'desc-preproc_bold:motion_correction_only_mcflirt'], ['FSL-AFNI-bold-ref:template_resample'], ['FSL-AFNI-brain-mask:template_resample'], ['FSL-AFNI-brain-probseg:template_resample'], 'space-bold_desc-brain_mask:bold_mask_fsl_afni'], ['pe-direction:func_metadata_ingress'], ['epi-1:fmap_ingress'], ['epi-1-pedir:fmap_pedir_ingress'], ['epi-1-dwell:fmap_dwell_ingress'], ['epi-1-total-readout:fmap_readout_ingress'], ['epi-2:fmap_ingress'], ['epi-2-pedir:fmap_pedir_ingress'], ['epi-2-dwell:fmap_dwell_ingress'], ['epi-2-total-readout:fmap_readout_ingress'], 'fsl-blip-warp:distcor_blip_fsl_topup'], ['T1w:anat_ingress'], [['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'desc-preproc_T1w:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-preproc_T1w:anatomical_init', 'space-T1w_desc-brain_mask:brain_mask_niworkflows_ants'], ['T1w:anat_ingress', 'desc-head_T1w:anatomical_init'], 'desc-preproc_T1w:brain_extraction'], ['T1w-brain-template-funcreg:template_resample'], ['T1w-brain-template-deriv:template_resample'], 'space-template_res-derivative_desc-bold_mask:single_step_resample_stc_timeseries_to_T1template'], 'CpacVariant': {'bold': ['apply_phasediff_to_timeseries_separately_single_step_resampling_from_stc', 'apply_blip_to_timeseries_separately_single_step_resampling_from_stc'], 'space-template_res-derivative_mask': ['single_step_resample_stc_timeseries_to_T1template']}, 'Sources': ['sbref', 'desc-stc_bold', 'motion-basefile', 'space-bold_desc-brain_mask', 'coordinate-transformation', 'from-T1w_to-template_mode-image_xfm', 'from-bold_to-T1w_mode-image_desc-linear_xfm', 'from-bold_to-template_mode-image_xfm', 'fsl-blip-warp', 'T1w', 'desc-preproc_T1w', 'T1w-brain-template-funcreg', 'T1w-brain-template-deriv'], 'Template': 'MNI152NLin6ASym - T1w-based skull-stripped template resampled to the desired functional derivative resolution - /usr/share/fsl/5.0/data/standard/MNI152_T1_2mm_brain.nii.gz', 'Resolution': '2mm'}
```

Reply:
I refactored this section to make a set of the resources in `CpacProvenance` and look for one that starts with `template_` if `Template` is not in `json`: https://github.com/FCP-INDI/C-PAC/blob/b2ac94f5ee2b9bd3a0266bb4f9a3cafb5325d66f/CPAC/pipeline/engine.py#L133-L148

First one found that isn't `FSL-AFNI-bold-ref` gets applied